test/integration/sagemaker/test_mnist.py

```diff
@@ -19,41 +19,54 @@
 
-from test.integration import mnist_script, model_cpu_dir
+from test.integration import model_cpu_dir, mnist_cpu_script, mnist_gpu_script, model_eia_dir, mnist_eia_script
 from test.integration.sagemaker.timeout import timeout_and_delete_endpoint
 
 
-@pytest.mark.skip_gpu
+@pytest.mark.cpu_test
 def test_mnist_distributed_cpu(sagemaker_session, ecr_image, instance_type):
     instance_type = instance_type or 'ml.c4.xlarge'
-    _test_mnist_distributed(sagemaker_session, ecr_image, instance_type)
+    model_dir = os.path.join(model_cpu_dir, 'model_mnist.tar.gz')
+    _test_mnist_distributed(sagemaker_session, ecr_image, instance_type, model_dir, mnist_cpu_script)
 
 
-@pytest.mark.skip_cpu
+@pytest.mark.gpu_test
 def test_mnist_distributed_gpu(sagemaker_session, ecr_image, instance_type):
     instance_type = instance_type or 'ml.p2.xlarge'
-    _test_mnist_distributed(sagemaker_session, ecr_image, instance_type)
+    model_dir = os.path.join(model_cpu_dir, 'model_mnist.tar.gz')
+    _test_mnist_distributed(sagemaker_session, ecr_image, instance_type, model_dir, mnist_gpu_script)
 
 
-def _test_mnist_distributed(sagemaker_session, ecr_image, instance_type):
-    model_dir = os.path.join(model_cpu_dir, 'model_mnist.tar.gz')
+@pytest.mark.eia_test
+def test_mnist_eia(sagemaker_session, ecr_image, instance_type, accelerator_type):
+    instance_type = instance_type or 'ml.c4.xlarge'
+    # Scripted model is serialized with torch.jit.save().
+    # Inference test for EIA doesn't need to instantiate model definition then load state_dict
+    model_dir = os.path.join(model_eia_dir, 'model_mnist.tar.gz')
+    _test_mnist_distributed(sagemaker_session, ecr_image, instance_type, model_dir, mnist_eia_script,
+                            accelerator_type=accelerator_type)
+
 
+def _test_mnist_distributed(sagemaker_session, ecr_image, instance_type, model_dir, mnist_script,
+                            accelerator_type=None):
     endpoint_name = sagemaker.utils.unique_name_from_base("sagemaker-pytorch-serving")
 
     model_data = sagemaker_session.upload_data(
         path=model_dir,
         key_prefix="sagemaker-pytorch-serving/models",
     )
 
-    pytorch = PyTorchModel(model_data,
-                           'SageMakerRole',
-                           mnist_script,
-                           image=ecr_image,
-                           sagemaker_session=sagemaker_session)
+    pytorch = PyTorchModel(model_data=model_data, role='SageMakerRole', entry_point=mnist_script,
+                           image=ecr_image, sagemaker_session=sagemaker_session)
 
     with timeout_and_delete_endpoint(endpoint_name, sagemaker_session, minutes=30):
-        predictor = pytorch.deploy(initial_instance_count=1, instance_type=instance_type,
-                                   endpoint_name=endpoint_name)
+        # Use accelerator type to differentiate EI vs. CPU and GPU. Don't use processor value
+        if accelerator_type is not None:
```

Comment:
what happens when you pass accelerator_type=None to pytorch.deploy? If that is the default argument, you can probably get rid of this if condition

Reply:
https://github.com/aws/sagemaker-python-sdk/blob/934fd90f69896336deead766b13297c5c49b4d78/src/sagemaker/tensorflow/serving.py#L189

Looks like it is None by default.

Good catch.